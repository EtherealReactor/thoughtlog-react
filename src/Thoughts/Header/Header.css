.header {
  composes: header from '../../Welcome/Header/Header.css';
  min-width: 100%;
  position: fixed;
  z-index: 99;
  top: 0;
  background-color: #6A3387;
}

.header__logo span{
  color: #A0A3A5;
}