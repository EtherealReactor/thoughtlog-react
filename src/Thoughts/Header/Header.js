import React from 'react';
import Styles from '../../Welcome/Header/Header.css';
import HeaderStyles from './Header.css';
import SideBar from '../SideBar/SideBar';

export class Header extends React.Component {
  render() {
    return (
      <React.Fragment>
        <header className={HeaderStyles.header}>
          <div >
            <h1 className={HeaderStyles.header__logo}>
              <span>Thought</span>
              <span>Log</span>
            </h1>
          </div>
          
          <div >
            <SideBar className={Styles.header__nav} handleNewClick={this.props.handleNewClick}/>
          </div>
        </header>
      </React.Fragment>
    );
  };
};

export default Header;