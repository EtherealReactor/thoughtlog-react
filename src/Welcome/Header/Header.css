.header {
  display: flex;
  justify-content: space-around;
  -ms-align-items: center;
  align-items: center;
  box-shadow: 0 0px 1rem grey;
  height: 7rem;;
}

.header__logo {
    color: #68308C;
}

.header__logo span:last-child {
    color: #895EA8;
}

.header__nav {
  width: 15%;
}

.header__list {
  display: flex;
  padding: 0;
  list-style: none;
  justify-content: space-evenly;
}

/* .headerContainer {
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.header {
    height: 7rem;
    box-shadow: 0 0px 1rem grey;
}

.header__logo {
    color: #68308C;
    display: inline-block;
}

.header__logo span:last-child {
    color: #895EA8;
}

.logo__name {
    margin: 0;
    padding: 0;
    line-height: 7rem;
    font-size: 2.5rem;
}

.header__content {
    width: 80%;
    margin: 0 auto;
}

.navigation {
    float: right;
    line-height: 7rem;
}

.navigation__list {
    list-style: none;
    padding: 0;
    margin: 0;
}

.navigation__item {
    float: left;
    margin-left: 1.5rem;
} */