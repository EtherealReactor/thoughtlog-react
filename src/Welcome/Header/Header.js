import React from 'react';
import TypographyStyles from '../../UI/CSS/Typography.css';
import Styles from './Header.css';
import { NavLink } from 'react-router-dom';

const Header = (props) => {
  return (
    <React.Fragment>
      <header className={Styles.header}>
        <div >
          <h1 className={Styles.header__logo}>
            <span>Thought</span>
            <span>Log</span>
          </h1>
        </div>

        <nav className={Styles.header__nav}>
          <ul className={Styles.header__list}>
            <li >
              <NavLink
                to="/sign_in"
                exact
                className={TypographyStyles.Link}
                activeClassName={TypographyStyles.Selected}>
                Log in
              </NavLink>                      
            </li>
            <li >
              <NavLink 
                to="/sign_up"
                className={TypographyStyles.Link}
                activeClassName={TypographyStyles.Selected}>
                Sign up
              </NavLink> 
            </li>
          </ul>
        </nav>
      </header>
    </React.Fragment>
  );
};

export default Header;