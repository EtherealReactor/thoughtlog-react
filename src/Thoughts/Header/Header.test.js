import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Header from './Header';

describe("Component === Thoughts === Header", () => {
  it("must match snapshot", () => {
    let wrapper = shallow(<Header />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should contain all the necessary dom", () => {
    let wrapper = shallow(<Header />);
    expect(wrapper.find('div')).toHaveLength(3);
    expect(wrapper.find('header')).toHaveLength(1);
    expect(wrapper.find('h1')).toHaveLength(1);
    expect(wrapper.find('span')).toHaveLength(2);
  });
});