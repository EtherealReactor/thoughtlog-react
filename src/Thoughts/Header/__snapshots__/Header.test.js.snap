// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component === Thoughts === Header must match snapshot 1`] = `
<div>
  <header>
    <div>
      <div>
        <h1>
          <span>
            Thought
          </span>
          <span>
            Log
          </span>
        </h1>
      </div>
    </div>
  </header>
</div>
`;
